pymc_marketing/clv/models/beta_geo.py

```diff
@@ -344,3 +344,61 @@ def expected_num_purchases_new_customer(
         return (left_term * right_term).transpose(
             "chain", "draw", "t", missing_dims="ignore"
         )
+
+    def probability_of_n_purchases_up_to_time(
+        self,
+        t: Union[np.ndarray, pd.Series],
+        n: int,
+    ):
+        r"""
+        Compute the probability of n purchases.
+         .. math::  P( N(t) = n | \text{model} )
+        where N(t) is the number of repeat purchases a customer makes in t
+        units of time.
+        Comes from equation (8) of [2]_.
+        Parameters
+        ----------
+        t: float
+            number units of time
+        n: int
+            number of purchases
+        Returns
+        -------
+        float:
+            Probability to have n purchases up to t units of time
+        References
+        ----------
+        .. [2] Fader, Peter S., Bruce G.S. Hardie, and Ka Lok Lee (2005a),
+        "Counting Your Customers the Easy Way: An Alternative to the
+        Pareto/NBD Model," Marketing Science, 24 (2), 275-84.
+        """
+
+        a, b, alpha, r = self._unload_params()
+
+        first_term = (
+            beta(a, b + n)
+            / beta(a, b)
+            * gamma(r + n)
+            / gamma(r)
+            / gamma(n + 1)
+            * (alpha / (alpha + t)) ** r
+            * (t / (alpha + t)) ** n
+        )
+
+        if n > 0:
+            j = np.arange(0, n)
+            finite_sum = 0
+            for j in range(n):
```

Comment:
Looping can be better using [`scan`s](https://pytensor.readthedocs.io/en/latest/library/scan.html). Mostly just pointing this out as an FYI for now as they can be daunting (I have not yet used `scan`s myself...)

Reply:
Aren't scans only needed for `pytensor` variables?

Reply:
> Aren't scans only needed for `pytensor` variables?

Ah yes, oops, that's correct! I was thinking of `pytensor.tensor.gammaln`...